buildSrc/subprojects/packaging/src/main/kotlin/org/gradle/gradlebuild/packaging/ShadedJarPlugin.kt

```diff
@@ -135,6 +139,29 @@ open class ShadedJarPlugin : Plugin<Project> {
         }
     }
 
+    private
+    fun Project.defaultShadedJarLocation() =
+        layout.buildDirectory.file("shaded-jar/${base.archivesBaseName}-shaded-${rootProject.extra["baseVersion"]}.jar")
+
+    private
+    fun Project.addInstallShadedJarTask(shadedJarTask: TaskProvider<ShadedJar>) {
+        val installPathProperty = "${project.name}ShadedJarInstallPath"
+        val taskName = "install${project.name.capitalize()}ShadedJar"
+        fun targetFile() = findProperty(installPathProperty)?.let { File(findProperty(installPathProperty) as String) }
```

Comment:
I have to create different types for this task because:

- I can't use `Callable<File?>` for `into` method because it will be evaulated before the task executes, a NPE will be thrown.
- I can't use `File?` for `into` method because kotlin doesn't accept it.

Reply:
Can't you just throw the validation error in the `Callable`?